ocaml/sdk-gen/powershell/autogen/src/Connect-XenServer.cs

```diff
@@ -212,40 +208,19 @@ protected override void ProcessRecord()
                     }
                     catch (WebException e)
                     {
-                        if (e.InnerException is HttpRequestException requestException)
+                        if (e.InnerException?.InnerException is CertificateValidationException ex)
                         {
-                            if (requestException.InnerException is CertificateChangeException certificateChangeException)
+                            if (ShouldContinue(ex.Message, ex.Caption))
                             {
-                                if (Force || ShouldContinue(string.Format(CERT_CHANGED, certificateChangeException.Fingerprint, certificateChangeException.OldFingerprint, certificateChangeException.Trusted), CERT_HAS_CHANGED_CAPTION))
-                                {
-                                    AddCertificate(certificateChangeException.Hostname, certificateChangeException.Fingerprint);
-                                    i--;
-                                    continue;
-                                }
-                                else
-                                {
-                                    ThrowTerminatingError(new ErrorRecord(certificateChangeException, "", ErrorCategory.AuthenticationError, Url[i])
-                                    {
-                                        ErrorDetails = new ErrorDetails($"New certificate fingerprint rejected. ({certificateChangeException.Fingerprint} - {certificateChangeException.Hostname}).")
-                                    });
-                                }
+                                AddCertificate(ex.Hostname, ex.Fingerprint);

```

Comment:
I think the check of `Force` here is not needed, because if it is true, the exceptions are not raised.